snuba/api.py

```diff
@@ -344,55 +348,45 @@ def sdk_distribution(validated_body, timer):
 if application.debug or application.testing:
     # These should only be used for testing/debugging. Note that the database name
     # is checked to avoid scary production mishaps.
-    assert settings.CLICKHOUSE_TABLE in ('dev', 'test')
 
-    _ensured = False
+    _ensured = {}
 
-    def ensure_table_exists(force=False):
+    def ensure_table_exists(dataset, force=False, drop=False):
         global _ensured
 
-        if not force and _ensured:
+        if not force and _ensured.get(dataset, False):
             return
 
-        from snuba.clickhouse import get_table_definition, get_test_engine
+        # only run table creation statements on non-prod datasets
+        if not dataset.PROD:
```

Comment:
Yeah, this is kinda supposed to be the same purpose as `if settings.CLICKHOUSE_TABLE == 'dev'` (ie simple protection against deleting prod data if the app is misconfigured) except slightly more broad. Basically instead of having separate settings for clickhouse table name, kafka topic, etc. the configuration should be all encapsulated in a `DataSet` like [this](https://github.com/getsentry/snuba/pull/294/files#diff-4748d36d4c99330e47b82e88f38f5319R1018) so that all you have to do is specify the dataset name and all the correct settings come with it.

I should change the cli/migrate and cli/perf scripts to be dataset aware though and not to use this table == 'dev' thing anymore.